src/native/libs/System.Native/pal_networking.c

```diff
@@ -3180,18 +3179,72 @@ int32_t SystemNative_SendFile(intptr_t out_fd, intptr_t in_fd, int64_t offset, i
         // For everything other than EINTR, bail.
         return SystemNative_ConvertErrorPlatformToPal(errno);
     }
-
 #else
-    // If we ever need to run on a platform that doesn't have sendfile,
-    // we can implement this with a simple read/send loop.  For now,
-    // we just mark it as not supported.
-    (void)outfd;
-    (void)infd;
-    (void)offset;
-    (void)count;
+    // Emulate sendfile using a simple read/send loop.
     *sent = 0;
-    errno = ENOTSUP;
-    return SystemNative_ConvertErrorPlatformToPal(errno);
+
+    // Save the original input file position and seek to the offset position
+    off_t inputFileOrigOffset = lseek(in_fd, 0, SEEK_CUR);
+    if (inputFileOrigOffset == -1 || lseek(in_fd, offtOffset, SEEK_SET) == -1)
+    {
+        return SystemNative_ConvertErrorPlatformToPal(errno);
+    }
+
+    // Allocate a buffer
+    int bufferLength = Min(count, 80 * 1024 * sizeof(char));
+    char* buffer = (char*)malloc(bufferLength);
+    if (buffer == NULL)
+    {
+        return SystemNative_ConvertErrorPlatformToPal(errno);
+    }
+
+    // Repeatedly read from the source and write to the destination
+    while (count > 0)
+    {
+        size_t numBytesToRead = Min(count, bufferLength);
+
+        // Read up to what will fit in our buffer.  We're done if we get back 0 bytes or read 'count' bytes
+        ssize_t bytesRead;
+        while ((bytesRead = read(in_fd, buffer, numBytesToRead)) < 0 && errno == EINTR);
+        if (bytesRead == -1)
+        {
+            free(buffer);
+            return SystemNative_ConvertErrorPlatformToPal(errno);
```

Comment:
Is `free` guaranteed to not change errno?

Reply:
no, good catch. I handled this in an earlier iteration but looks like I simplified too much :)